packages/lesswrong/components/hooks/usePostsListView.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { FC, ReactNode, createContext, useContext, useState } from "react";
+import { TupleSet, UnionOf } from "../../lib/utils/typeGuardUtils";
+
+const postsListViewTypes = new TupleSet(["list", "card"] as const);
+
+export type PostsListViewType = UnionOf<typeof postsListViewTypes>;
+
+export const isPostsListViewType = (value: string): value is PostsListViewType =>
+  postsListViewTypes.has(value);
+
+type PostsListViewContext = {
+  view: PostsListViewType,
+  setView: (view: PostsListViewType) => void,
+}
+
+const postsListViewContext = createContext<PostsListViewContext>({
+  view: "list",
+  // eslint-disable-next-line
+  setView: () => console.error("Can't set view outside of PostsListViewProvider"),
+});
+
+export const PostsListViewProvider: FC<{children: ReactNode}> = ({children}) => {
+  const [view, setView] = useState<PostsListViewType>("list");
```

Comment:
Ok this is fixed now, although I think it would be good to make it a user setting in addition to a cookie. Because cookies tend to get cleared (and also it makes the analytics a bit easier)

Reply:
Personally I think it's better with just a cookie - I think it's very plausible that someone might want to use card view on desktop but list view on mobile for instance.

Maybe @agnesstenlund has an opinion?

Reply:
My hunch is a stored user setting for desktop and mobile independently would be ideal, if that's a thing? (agree both that cookies get cleared easily _and_ that people are pretty likely to want different settings on desktop and mobile)

Reply:
if it's a user setting does that mean logged out users wouldn't be able to change it? Or does it just mean we extra store it if you're logged in (and everyone gets the cookie)

Reply:
I generally lean towards user settings for the reasons Will mentioned. I'm guessing the hydration error is related to the logged out user cache. I'm not sure how you all want to handle the cache here - I feel like not using it is a big cost, and I find the layout shift from list to card view a bit jarring, so maybe we start with this feature just being an option for logged in users?